dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -303,30 +303,122 @@ def test_report(self):
             [
                 "samples_used", "column_count", "row_count", 
                 "row_has_null_ratio", 'row_is_null_ratio',
-                "unique_row_ratio", "duplicate_row_count", "file_type", "encoding"
+                "unique_row_ratio", "duplicate_row_count", "file_type",
+                "encoding", "profile_schema"
             ]
         )
         flat_report = self.trained_schema.report(report_options={"output_format":"flat"})
         self.assertEqual(test_utils.get_depth(flat_report), 1)
         with mock.patch('dataprofiler.profilers.helpers.report_helpers._prepare_report') as pr_mock:
             self.trained_schema.report(report_options={"output_format":'pretty'})
-            self.assertEqual(pr_mock.call_count, 2)
+            # Once for global_stats, once for each of 16 columns
+            self.assertEqual(pr_mock.call_count, 17)
+
+    def test_report_data_stats_accessible(self):
+        data = pd.DataFrame([[1, 2, 3, 4, 5, 6],
+                             [10, 20, 30, 40, 50, 60]],
+                            columns=["a", "b", "a", "b", "c", "d"])
+        profiler_options = ProfilerOptions()
+        profiler_options.set({'data_labeler.is_enabled': False})
+        profiler = dp.StructuredProfiler(data=data, options=profiler_options)
+        report = profiler.report()
+        schema = report["global_stats"]["profile_schema"]
+        data_stats = report["data_stats"]
+        expected_schema = {"a": [0, 2], "b": [1, 3], "c": [4], "d": [5]}
+        self.assertDictEqual(expected_schema, schema)
+        for name in schema:
+            for idx in schema[name]:
+                col_min = data.iloc[0, idx]
+                col_max = data.iloc[1, idx]
+                col_sum = col_min + col_max
+                self.assertEqual(name, data_stats[idx]["column_name"])
+                self.assertEqual(col_min, data_stats[idx]["statistics"]["min"])
+                self.assertEqual(col_max, data_stats[idx]["statistics"]["max"])
+                self.assertEqual(col_sum, data_stats[idx]["statistics"]["sum"])
+
+    def test_pretty_report_doesnt_cast_schema(self):
+        report = self.trained_schema.report(
+            report_options={"output_format": "pretty"})
+        self.assertIsInstance(report["global_stats"]["profile_schema"], dict)
+        self.assertIsInstance(report["data_stats"], list)
+        for idx_list in report["global_stats"]["profile_schema"].values():
+            self.assertIsInstance(idx_list, list)
+            for idx in idx_list:
+                self.assertIsInstance(idx, int)
+                self.assertTrue(idx >= 0)
```

Comment:
instead would it be better to just assert the dict?

Reply:
Fixed